### PR TITLE
feat: [CDS-13678]: Terraform artifactory added

### PR DESCRIPTION
**Pull request by willd0yle**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by kapilkumawat86** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx [hunk 1]
```diff
@@ -271,34 +291,41 @@ export default function TfVarFileList(props: TfVarFileProps): React.ReactElement
...
+                            allowableTypes={allowableTypes}
+                            name={getString('cd.varFileDetails')}
+                            onSubmitCallBack={(values: RemoteVar) => onSubmit(values, arrayHelpers)}
+                          />
+                        ) : (
+                          <TFRemoteWizard
+                            name={getString('cd.varFileDetails')}
+                            onSubmitCallBack={(values: RemoteVar) => onSubmit(values, arrayHelpers)}
+                            isEditMode={isEditMode}
+                            // initialValues={remoteInitialValues}
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx:317]: Initial values not required?

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    return connectorValue.connector.identifier
+  }
+
+  return connectorValue
+}
+
+export const terraformArtifactorySchema = (isConfig: boolean, getString: any) => {
+  const artifacts = {
+    repositoryName: Yup.string().required(getString('cd.artifactFormErrors.repositoryName'))
+    // artifactPaths: Yup.array().required(getString('cd.pathCannotBeEmpty'))
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts:97]: Remove commented code

Comment (kapilkumawat86) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts:97]: this need to be fixed not removed

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx [hunk 3]
```diff
@@ -107,16 +138,143 @@ export default function TerraformEditView(
...
+            type={connectorType}
+            name={getString('details')}
+            isEditMode={isEditMode}
+            connectorInfo={undefined}
+          />
+        ) : null}
+        {connectorType === Connectors.GIT ? (
+          <StepGitAuthentication
+            name={getString('credentials')}
+            onConnectorCreated={() => {
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:181]: Remove unhandled events

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/InputSteps/TFRemoteSection.tsx [hunk 4]
```diff
@@ -5,50 +5,107 @@
...
+  })
+
+  useEffect(() => {
+    if (ArtifactRepoError) {
+      showError(ArtifactRepoError.message)
+    }
+  }, [ArtifactRepoError])
+
+  useEffect(() => {
+    if (connectorVal && !ArtifactRepoData) {
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/InputSteps/TFRemoteSection.tsx:97]: we should add check, if store is Artifact, then only make this call and make this change for all places

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts (hunk 2 above)
Comment (kapilkumawat86) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts:97]: example     artifactPaths: Yup.lazy((value): Yup.Schema<unknown> => {
      if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
        return Yup.array().of(
          Yup.object().shape({
            path: Yup.string().min(1).required(getString('cd.pathCannotBeEmpty'))
          })
        )
      }
      return Yup.string().required(getString('cd.pathCannotBeEmpty'))
    })

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarStore.tsx [hunk 5]
```diff
@@ -85,22 +75,31 @@ export const TFVarStore: React.FC<StepProps<any> & TFVarStoreProps> = ({
...
-    <Layout.Vertical spacing="xxlarge" padding="small" className={css.tfVarStore}>
-      <Heading level={2} style={{ color: Color.GREY_800, fontSize: 24 }} margin={{ bottom: 'large' }}>
+    <Layout.Vertical padding="small" className={css.tfVarStore}>
+      <Heading level={2} style={{ color: Color.BLACK, fontSize: 24, fontWeight: 'bold' }} margin={{ bottom: 'large' }}>
         {getString('cd.specifyTfVarStore')}
       </Heading>
 
       <Layout.Horizontal flex={{ justifyContent: 'flex-start', alignItems: 'flex-start' }}>
-        {allowedTypes.map(item => (
+        {AllowedTypes.map(item => (
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarStore.tsx:95]: Use ThumbnailSelect under formik

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+        {AllowedTypes.map(item => (
+          <div key={item} className={css.squareCardContainer}>
+            <Card
+              className={css.connectorIcon}
+              selected={item === selectedConnector}
+              data-testid={`varStore-${item}`}
+              onClick={() => {
+                setSelectedConnector(item as ConnectorTypes)
+              }}
+            >
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:134]: Clear selected connector on change

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+    orgIdentifier: string
+    accountId: string
+  }>()
+
+  useEffect(() => {
+    const selectedStore = isTerraformPlan
+      ? data?.spec?.configuration?.configFiles?.store?.type
+      : data?.spec?.configuration?.spec?.configFiles?.store?.type
+    setSelectedConnector(selectedStore)
+  }, [])
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:85]: updateDependencies here

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+            <Text color={Color.BLACK_100}>{item}</Text>
+          </div>
+        ))}
+      </Layout.Horizontal>
+
+      <Formik
+        formName="tfPlanConfigForm"
+        enableReinitialize={true}
+        onSubmit={() => {
+          /* istanbul ignore next */
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:146]: Why no submit here

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+              </Layout.Horizontal>
+            </Form>
+          )
+        }}
+      </Formik>
+    </Layout.Vertical>
+  )
+}
+
+interface TerraformConfigStepTwoProps {
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:231]: Split this file in two separate components

**Comment by kapilkumawat86** [comment]
Also rebase the PR

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+        configuration: Yup.object().shape({
+          ...configSetup
+        })
+      })
+    })
+  }
+
+  return Yup.object().shape({
+    varFile: Yup.object().shape({
+      identifier: Yup.string().required(getString('common.validation.identifierIsRequired')),
```
Comment (kapilkumawat86) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts:122]: Use this over here https://github.com/wings-software/nextgenui/blob/72d5bd71b0c9689787afae077005137c86a082d4/src/modules/10-common/utils/Validation.tsx#L52

Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryFormHelper.ts:131]: Done

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx (hunk 9 above)
Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:231]: Done

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx (hunk 8 above)
Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:146]: Will move to from button onclick to the submit function

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx (hunk 3 above)
Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:181]: They are a required prop for the component, if they are removed we get typescript errors

**Review by kapilkumawat86 — DISMISSED** [review]

**Comment by kapilkumawat86** [comment]
retrigger jest

**Pull request merged**